src/lib/lore.sh

```diff
@@ -458,3 +487,167 @@ function download_series()
   done
   wait
 }
+
+# This function deletes a patch series from the local storage
+#
+# @download_dir_path: The path to the directory where the series was stored
+# @patch_title: The title of the patch
+# @flag: Flag to control function output
+#
+# Return:
+# Return 0 if there are files related to the patch to be deleted and 2 (ENOENT)
+# if there aren't.
+function delete_series_from_local_storage()
+{
+  local download_dir_path="$1"
+  local patch_title="$2"
+  local flag="$3"
+  local filename_pattern
+  local files_to_remove
+  local ret
+
+  flag=${flag:-'SILENT'}
+
+  filename_pattern=$(convert_title_to_patch_name "$patch_title")
+
+  # Find doesn't return a non-zero status code when not matching any file.
+  # We use grep to work-around it.
+  files_to_remove=$(find "${download_dir_path}" -iname "*${filename_pattern}" | grep '.')
+  ret="$?"
+
+  if [[ "$ret" == 0 ]]; then
+    while IFS=' ' read -r file; do
+      is_safe_path_to_remove "$file"
+      if [[ "$?" == 0 ]]; then
+        cmd_manager "$flag" "rm ${file}"
+      fi
+    done <<< "${files_to_remove}"
+  else
+    return 2 # ENOENT
+  fi
+}
+
+# This function creates the lore bookmarked series file if it doesn't
+# already exists
+function create_lore_bookmarked_file()
+{
+  create_lore_data_dir
```

Comment:
In this new implementation, the above function can fail. Please check if it succeeds before creating a new file.